build/resolver/git/entrypoint.sh

```diff
@@ -177,46 +169,37 @@ parseRevision() {
 }
 parseRevision
 
-parseCloneRes() {
-    local clone_log=$1
-    local clone_res=$2
-    cat $clone_log
-    if [ $clone_res -ne 0 ]; then
-        if grep "The requested URL returned error: 500" $clone_log > /dev/null 2>&1; then
-            # use code 99 for git clone --depth fail
-            rm -f $clone_log
-            return 99
-        else
-            exit $clone_res
-        fi
-    fi
-    return 0
+retryGitOpWithoutDepthFlagIfFromBitbucket() {
+  local op="$1"
+  shift
+  if (! git "$op" ${GIT_DEPTH_OPTION:-} "$@") && [[ ${SCM_TYPE} == 'Bitbucket' ]]; then
+    echo "Fail to pull with depth flag, retry without depth flag..."
+    git "$op" "$@"
+  fi
 }
 
 pull() {
-    local parse_res
-    local clone_res
     git config --global http.sslVerify false
     git config --global http.postBuffer 500M
     NO_AUTH_SCM_URL=${SCM_URL}
     # If data existed and pull policy is IfNotPresent, perform incremental pull.
-    if [ -e $WORKDIR/data ] && [ ${PULL_POLICY:=Always} == "IfNotPresent" ]; then
-        cd $WORKDIR/data
+    if [ -e "$WORKDIR/data" ] && [ ${PULL_POLICY:=Always} == "IfNotPresent" ]; then
+        cd "$WORKDIR/data"
         # Ensure existed data come from the git repo
-        git remote -v | grep ${SCM_URL##*//} || {
+        git remote -v | grep "${SCM_URL##*//}" || {
             echo "Existed data not a valid git repo for ${SCM_URL##*//}"
             exit 1
         }
 
         echo "Fetch $SCM_REVISION from origin"
-        git fetch -v ${GIT_DEPTH_OPTION:-} origin $SCM_REVISION
+        retryGitOpWithoutDepthFlagIfFromBitbucket fetch -v origin "$SCM_REVISION"
```

Comment:
I think the function name should be renamed normally like cloneRetry blablabla, or something like that. And comment bitbucket.

Reply:
IMHO it is not quite convenient to view the comment of a function in bash, and I think it might be more obvious to user to use an informative(but maybe verbose) function name.